File: Graph/BFS.c
#include<stdio.h>
int main()
{
	int v,e,i,j,v1,v2,k=-1,root;
	printf("enter no of (vertices,edges) : ");
	scanf("%d%d",&v,&e);
	printf("enter edges( , ) between (0-%d) : ",v-1);
	int visited[v]={0},queued[v],A[v][v];
	for(i=0;i<e;i++)
	{
		scanf("%d%d",&v1,&v2);
		A[v1][v2]=1;
	}
	printf("select root node :");
	scanf("%d",&root);
	visited[root]=1;
	queued[++k]=root;

	for(j=0;j<=k;j++)
	{
	 for(i=0;i<v;i++)
     {
       if(A[queued[j]][i]==1&&visited[i]==0)
	   {	
	       queued[++k]=i;
	       visited[i]=1;
	   }
	  
	 }	 
	  printf("%d ",queued[j]);
    }
   return 0;
}

File: Graph/BFS.java
import java.util.Scanner;
import java.util.*;
import java.io.*;
public class BFS
{
	public static void main(String args[])
	{
		int i,v,e,e1,e2;
		Scanner scan=new Scanner(System.in);
		System.out.println("enter no of vertices and edges:");
		 v=scan.nextInt();
		 e=scan.nextInt();
		
		Graph g=new Graph(v);
		for(i=0;i<e;i++)
		{
			System.out.print("enter edges "+(i+1)+" : ");
			e1=scan.nextInt();
			e2=scan.nextInt();
			g.addEdge(e1,e2);
		}
		System.out.println("enter initial vertice : ");
		 int iv=scan.nextInt();
		 g.bfs(iv);
	}
}

 class Graph
{ 
   private int V;
   private LinkedList<Integer> adj[];   
	  Graph(int v)
	  {
		V=v;
		adj=new LinkedList[v];
		for(int i=0;i<v;i++)
		   adj[i]=new LinkedList<Integer>();
	  }	   
		   void addEdge(int e1,int e2)
		  {
			  adj[e1].add(e2);
		  }
	   void bfs(int iv)
	   {
		   boolean visited[]=new boolean[V];
		   LinkedList<Integer> queue= new LinkedList<Integer>();
		   visited[iv]=true;
		   queue.add(iv);
		   while(queue.size()!=0)
		   {
			   int s=queue.poll();
			   System.out.print(s+" ");
			   Iterator<Integer> i=adj[s].listIterator();
			   while(i.hasNext())
			   {
				   int n=i.next();
				   if(!visited[n])
				   {
					   visited[n]=true;
					   queue.add(n);
				   }
			   }
		   }
	   }
}
     	